Makefile

```diff
@@ -69,27 +69,57 @@ essr: VERSION
 
 
 ## the rest of the targets are for ESS developer's use only :
+.PHONY: tarballs
+tarballs: ess-$(ESSVERSION).tar ess-$(ESSVERSION).tgz ess-$(ESSVERSION).zip # TODO: ess-plus-$(VERSION).tar
 
-# Create .tgz and .zip files only
-# GNUTAR=gtar make tarballs
-tarballs: $(ESSDIR)
-	@echo "**********************************************************"
-	@echo "** Making distribution of ESS for release $(ESSVERSION) from $(ESSDIR)/"
-	@echo "** Making pdf and html documentation"
-#	the making of pdf, html, ESSR.rds shall go into `make dist' #752
-	@cd $(ESSDIR)/doc/ ; $(MAKE) pdf
-	@cd $(ESSDIR)/doc/ ; $(MAKE) html
+.PHONY: tgz
+tgz: ess-$(ESSVERSION).tgz
+ess-$(ESSVERSION).tgz: $(ESSDIR)
 	@echo "** Creating .tgz file **"
 	test -f $(ESSDIR).tgz && rm -rf $(ESSDIR).tgz || true
 	$(GNUTAR) hcvofz $(ESSDIR).tgz $(ESSDIR)
 	@echo "Signing tgz file"
 	$(GPG) -ba -o $(ESSDIR).tgz.sig $(ESSDIR).tgz
+
+.PHONY: zip
+zip: ess-$(ESSVERSION).zip
+ess-$(ESSVERSION).zip: $(ESSDIR)
 	@echo "** Creating .zip file **"
 	test -f $(ESSDIR).zip && rm -rf $(ESSDIR).zip || true
 	zip -r $(ESSDIR).zip $(ESSDIR)
 	@echo "Signing zip file"
 	$(GPG) -ba -o $(ESSDIR).zip.sig $(ESSDIR).zip
-	touch $@
+
+.PHONY: package
+package: ess-$(ESSVERSION).tar
+ess-$(ESSVERSION).tar: ess-pkg.el
+	@echo "Creating $@"
```

Comment:
There is a new ./targets folder for make utilities. I would prefer that we move this into bash script there; it's much nicer to develop. Same for the following elisp. I think most of this stuff can be done at elisp level actually, but fine either way I guess.

Reply:
Yeh, those vars are problematic. Let's leave the tar as it is. 

The pkg.el creation should be in a proper lisp script though. It should take ees.el and parse metadata from there. Functionality from packae.el could be re-used but I think it's straightforward either way.

Reply:
OK, done. There's a new script in targets/